homeassistant/components/evohome/__init__.py

```diff
@@ -486,24 +487,25 @@ async def _update_v2(self, *args, **kwargs) -> None:
         except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
             _handle_exception(err)
         else:
-            async_dispatcher_send(self.hass, DOMAIN)
-
             _LOGGER.debug("Status = %s", status)
 
         if access_token != self.client.access_token:
             await self.save_auth_tokens()
 
-    async def async_update(self, *args, **kwargs) -> None:
+    async def async_update(self, *args, delay=None, **kwargs) -> None:
         """Get the latest state data of an entire Honeywell TCC Location.
 
         This includes state data for a Controller and all its child devices, such as the
         operating mode of the Controller and the current temp of its children (e.g.
         Zones, DHW controller).
         """
-        await self._update_v2()
+        if delay:
+            await asyncio.sleep(delay)
+
+        elif self.client_v1:
+            await self._update_v1_api()
 
-        if self.client_v1:
-            await self._update_v1()
+        await self._update_v2_api()
```

Comment:
`self._update_v1_api()` is not required for a ad-hoc refresh, only every `scan_interval`

Reply:
It is a bit unintuitive that `delay` changes the refresh behavior.  Can we call the argument something that shows the intent like `manual_refresh` since the only use case is a delay of 1s?

Reply:
OK, a few tweaks, `manual_refresh` and `update_broker` (there is a `force_refresh` in there somewhere too) - I changed another parameter name to make things more clear.